src/backend/executor/nodeValuesscan.c

```diff
@@ -167,12 +167,12 @@ ValuesNext(ValuesScanState *node)
 		ExecStoreVirtualTuple(slot);
 
         /* CDB: Label each row with a synthetic ctid for subquery dedup. */
-        if (node->cdb_want_ctid)
+        if (node->cdb_want_ctid && !TupIsNull(slot))
         {
             HeapTuple   tuple = ExecFetchSlotHeapTuple(slot); 
 
-            ItemPointerSet(&tuple->t_self, node->curr_idx >> 16,
-                           (OffsetNumber)node->curr_idx);
+			ItemPointerData cdb_fake_ctid = { node->curr_idx >> 16, (OffsetNumber)node->curr_idx };
+			slot_set_ctid_from_fake(slot, &cdb_fake_ctid);
```

Comment:
The patch is "incrementing" a throw-away ctid repeatedly, which is a bit sketchy. Shouldn't we simply offset the current counter by 1?

```suggestion
			ItemPointerSet(&tuple->t_self, node->curr_idx >> 16,
						   (OffsetNumber) (node->curr_idx+1));
```

Reply:
yse, you can use this

Reply:
node->curr_idx+1 may will overflow?